finetune/lora.py

```diff
@@ -119,25 +118,31 @@ def main(fabric: L.Fabric, data_dir: Path, checkpoint_dir: Path, out_dir: Path,
     )
     checkpoint_path = checkpoint_dir / "lit_model.pth"
     fabric.print(f"Loading model {str(checkpoint_path)!r} with {config.__dict__}")
-    with fabric.init_module(empty_init=False), quantization(quantize):
+    with fabric.init_module(empty_init=(devices > 1)), quantization(quantize):
         model = GPT(config)
-    with lazy_load(checkpoint_path) as checkpoint:
```

Comment:
Note that this still has the disadvantage that `lazy_load` is not used anymore with `devices=1`. Perhaps we should have separate loading paths for each case until Fabric supports lazy loading for the other strategies

Reply:
Roughly:

```python
if devices == 1:
    with lazy_load():
        model.load_state_dict()

fabric.setup_module()


if devices > 1:
    fabric.load_raw(checkpoint_path, model, strict=False)
```

Reply:
Do you think this is worth it for this application? For inference the load time is noticeable yes and lazy_load is preferred, but for finetuning that runs for a long time, the load time is insignificant. 

We can certainly check whether lazy-loading is reliable enough to have it enabled in general.

Reply:
`lazy_load` is advantageous not only for reducing loading time but also for limiting peak memory usage. Since these scripts target consumer setups, I would suggest that we keep it for those with limited system RAM compared to VRAM.